Documents/Pseudocode/Pseudocode_Final

```diff
@@ -254,17 +258,16 @@ Class Network
 	void classify() 				//we don't need any parameters here, because constructor stores the name of validation file in the training_data_file
 	{
 		//int ambiguous_data (0);
-		//int numTrainingInputs;		//will tell us the number of the training inputs in the verfification file
+		//int numTrainingInputs;		//will tell us the number of the training inputs in the verfification file	//consider numInputs [-Ami]
 		//int biggest;
 		//to calculate numTrainingInputs, we can call tellg(), which will give us the size of file in bytes so that 
 		//numTrainingInputs = size of file/size of a single verification datum
-		//Loop index i from 0 to size of (verification data) //from i = 0 to end
-			//call forwardPass(i), which takes a datum via training_data_filename and computes the activations <-- is this the right file?
+		//Loop index i from 0 to size of (verification data) //from i = 0 to numInput [-Ami]			//call forwardPass(i), which takes a datum via training_data_filename and computes the activations <-- is this the right file?
 			//set biggest = 0, which will act as the index of the biggest activation value
 			//Looping index j from 1 to size of (activations[num_layers]) (j=0 is assumed to be biggest)
 				//if activations(num_layers, j) > activations(num_layers, biggest) 
 					//then biggest = j
-				//else if activations(num_layers, j) == activations(num_layers, biggest)
+				//else if activations(num_layers, j) == activations(num_layers, biggest) 	//consider the ouput [0.1,0.1,0.2] [-Ami]
```

Comment:
Ay, muy correcto. I checked the history, but wouldn't the previous algorithm increment ambiguous_data for 
[0.1, 0.1, 0.1, 0.2]? 

Perhaps we could do an initial pass to assign the largest number, then a secondary to check whether or not that number appears twice:
//Looping index j from 1 to activations[num_layers].nc()
&nbsp;&nbsp;&nbsp;&nbsp;//if activations[num_layers](j, 1) > activations[num_layers](biggest, 1)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//biggest = j
//Looping index j from 0 to activations[num_layers].nc()
&nbsp;&nbsp;&nbsp;&nbsp;//if activations[num_layers](j, 1) == activations[num_layers](biggest, 1)
&nbsp;&nbsp;&nbsp;&nbsp;//equalityCounter++
//If equalityCounter > 1
&nbsp;&nbsp;&nbsp;&nbsp;//increment ambiguous_data

Or use a flag:
//bool doubleCheck = false
//Looping index j from 1 to activations[num_layers].nc()
&nbsp;&nbsp;&nbsp;&nbsp;//if activations[num_layers](j, 1) > activations[num_layers](biggest, 1)
&nbsp;&nbsp;&nbsp;&nbsp;//biggest = j
&nbsp;&nbsp;&nbsp;&nbsp;//else if activations[num_layers](j, 1) == activations[num_layers](biggest, 1)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//doubleCheck = true
//If doubleCheck
&nbsp;&nbsp;&nbsp;&nbsp;//Looping index j from 1 to activations[num_layers].nc()
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//if activations[num_layers](i, 1) == activations[num_layers](biggest, 1)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//ambiguous_data++
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//break loop <-- bad?

Anyway we end up doing it, it should work for arbitrarily-sized output layers.

Reply:
I think we can think about this once we start coding! We will be getting more on how the code handles the cases like this.